File: app/src/main/java/com/example/appctruyn/LibraryBottomSheet.java
package com.example.appctruyn;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.Switch;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.example.appctruyn.model.LibraryStory;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

public class LibraryBottomSheet extends BottomSheetDialogFragment {

    public interface OnDeleteClickListener {
        void onDeleteClick();
    }

    public interface OnBookmarkToggleListener {
        void onBookmarkToggle(boolean isChecked);
    }

    private OnDeleteClickListener onDeleteClick;
    private OnBookmarkToggleListener onBookmarkToggle;

    private static final String ARG_STORY_ID = "storyId";
    private static final String ARG_TITLE = "title";
    private static final String ARG_COVER = "coverUrl";
    private static final String ARG_LAST_CHAP = "lastChap";
    private static final String ARG_TOTAL_CHAP = "totalChap";
    private static final String ARG_NOTIFY = "notifyEnabled";

    public static LibraryBottomSheet newInstance(LibraryStory item) {
        LibraryBottomSheet fragment = new LibraryBottomSheet();
        Bundle args = new Bundle();
        args.putString(ARG_STORY_ID, item.getStoryId());
        args.putString(ARG_TITLE, item.getTitle());
        args.putString(ARG_COVER, item.getCoverUrl());
        args.putInt(ARG_LAST_CHAP, item.getLastChap());
        args.putInt(ARG_TOTAL_CHAP, item.getTotalChap());
        args.putBoolean(ARG_NOTIFY, item.isNotifyEnabled());
        fragment.setArguments(args);
        return fragment;
    }

    public void setOnDeleteClickListener(OnDeleteClickListener listener) {
        this.onDeleteClick = listener;
    }

    public void setOnBookmarkToggleListener(OnBookmarkToggleListener listener) {
        this.onBookmarkToggle = listener;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.bottom_sheet_library, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        String title = getArguments() != null ? getArguments().getString(ARG_TITLE) : "";
        String coverUrl = getArguments() != null ? getArguments().getString(ARG_COVER) : "";
        boolean notifyEnabled = getArguments() != null && getArguments().getBoolean(ARG_NOTIFY);

        ImageView ivCover = view.findViewById(R.id.ivCover);
        TextView tvTitle = view.findViewById(R.id.tvTitle);
        View btnDownload = view.findViewById(R.id.btnDownload);
        View btnDelete = view.findViewById(R.id.btnDelete);
        Switch switchNotify = view.findViewById(R.id.switchNotify);

        tvTitle.setText(title);
        switchNotify.setChecked(notifyEnabled);

        Glide.with(requireContext()).load(coverUrl)
                .placeholder(R.drawable.ic_launcher_background)
                .into(ivCover);

        btnDownload.setOnClickListener(v -> {
            // Tính năng tải truyện (sắp ra mắt)
            dismiss();
        });

        btnDelete.setOnClickListener(v -> {
            if (onDeleteClick != null) {
                onDeleteClick.onDeleteClick();
            }
            dismiss();
        });

        switchNotify.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (onBookmarkToggle != null) {
                onBookmarkToggle.onBookmarkToggle(isChecked);
            }
        });
    }
}

File: app/src/main/java/com/example/appctruyn/LibraryBottomSheet.kt
package com.example.appctruyn

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.Switch
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.appctruyn.model.LibraryStory
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

class LibraryBottomSheet : BottomSheetDialogFragment() {

    var onDeleteClick: (() -> Unit)? = null
    var onBookmarkToggle: ((Boolean) -> Unit)? = null

    companion object {
        private const val ARG_STORY_ID = "storyId"
        private const val ARG_TITLE = "title"
        private const val ARG_COVER = "coverUrl"
        private const val ARG_LAST_CHAP = "lastChap"
        private const val ARG_TOTAL_CHAP = "totalChap"
        private const val ARG_NOTIFY = "notifyEnabled"

        fun newInstance(item: LibraryStory): LibraryBottomSheet {
            val f = LibraryBottomSheet()
            f.arguments = Bundle().apply {
                putString(ARG_STORY_ID, item.storyId)
                putString(ARG_TITLE, item.title)
                putString(ARG_COVER, item.coverUrl)
                putInt(ARG_LAST_CHAP, item.lastChap)
                putInt(ARG_TOTAL_CHAP, item.totalChap)
                putBoolean(ARG_NOTIFY, item.notifyEnabled)
            }
            return f
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? = inflater.inflate(R.layout.bottom_sheet_library, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val title = arguments?.getString(ARG_TITLE) ?: ""
        val coverUrl = arguments?.getString(ARG_COVER) ?: ""
        var notifyEnabled = arguments?.getBoolean(ARG_NOTIFY) ?: false

        val ivCover = view.findViewById<ImageView>(R.id.ivCover)
        val tvTitle = view.findViewById<TextView>(R.id.tvTitle)
        val btnDownload = view.findViewById<View>(R.id.btnDownload)
        val btnDelete = view.findViewById<View>(R.id.btnDelete)
        val switchNotify = view.findViewById<Switch>(R.id.switchNotify)

        tvTitle.text = title
        switchNotify.isChecked = notifyEnabled

        Glide.with(requireContext()).load(coverUrl)
            .placeholder(R.drawable.ic_launcher_background)
            .into(ivCover)

        btnDownload.setOnClickListener {
            // Tính năng tải truyện (sắp ra mắt)
            dismiss()
        }

        btnDelete.setOnClickListener {
            onDeleteClick?.invoke()
            dismiss()
        }

        switchNotify.setOnCheckedChangeListener { _, isChecked ->
            notifyEnabled = isChecked
            onBookmarkToggle?.invoke(isChecked)
        }
    }
}